plugins/woocommerce/tests/php/includes/class-wc-tracker-test.php

```diff
@@ -305,4 +305,54 @@ public function test_get_tracking_data_admin_install_timestamp() {
 	public function test_get_tracking_data_snapshot_generation_time() {
 		$this->assertGreaterThan( 0, WC_Tracker::get_tracking_data()['snapshot_generation_time'] );
 	}
+
+	/**
+	 * @testDox Test woocommerce_allow_tracking related data is included in tracking snapshot.
+	 */
+	public function test_tracking_data_woocommerce_allow_tracking() {
+		$current_woocommerce_allow_tracking = get_option( 'woocommerce_allow_tracking', 'no' );
+
+		// Clear everything.
+		update_option( 'woocommerce_allow_tracking', 'no' );
+		delete_option( 'woocommerce_allow_tracking_last_modified' );
+		delete_option( 'woocommerce_allow_tracking_first_optin' );
+
+		$tracking_data = WC_Tracker::get_tracking_data();
+		$this->assertArrayHasKey( 'woocommerce_allow_tracking', $tracking_data );
+		$this->assertArrayHasKey( 'woocommerce_allow_tracking_last_modified', $tracking_data );
+		$this->assertArrayHasKey( 'woocommerce_allow_tracking_first_optin', $tracking_data );
+
+		$this->assertEquals( $tracking_data['woocommerce_allow_tracking'], 'no' );
+		$this->assertEquals( $tracking_data['woocommerce_allow_tracking_last_modified'], 'unknown' );
+		$this->assertEquals( $tracking_data['woocommerce_allow_tracking_first_optin'], 'unknown' );
+
+		$time_one = time();
+		update_option( 'woocommerce_allow_tracking', 'yes' );
+		$tracking_data = WC_Tracker::get_tracking_data();
+		$this->assertEquals( $tracking_data['woocommerce_allow_tracking'], 'yes' );
+		$this->assertTrue( $tracking_data['woocommerce_allow_tracking_last_modified'] >= $time_one );
+		$this->assertTrue( $tracking_data['woocommerce_allow_tracking_first_optin'] >= $time_one );
+
+		sleep( 1 ); // be sure $time_two is at least one second after $time_one.
+		$time_two = time();
+		update_option( 'woocommerce_allow_tracking', 'no' );
+		$tracking_data = WC_Tracker::get_tracking_data();
+
+		$this->assertEquals( $tracking_data['woocommerce_allow_tracking'], 'no' );
+		$this->assertTrue( $tracking_data['woocommerce_allow_tracking_last_modified'] >= $time_two );
+		$this->assertTrue( $tracking_data['woocommerce_allow_tracking_first_optin'] >= $time_one && $tracking_data['woocommerce_allow_tracking_first_optin'] < $time_two );
+
+		sleep( 1 ); // be sure $time_three is at least one second after $time_two.
+		$time_three = time();
+		update_option( 'woocommerce_allow_tracking', 'yes' );
+		$tracking_data = WC_Tracker::get_tracking_data();
+		$this->assertEquals( $tracking_data['woocommerce_allow_tracking'], 'yes' );
+		$this->assertTrue( $tracking_data['woocommerce_allow_tracking_last_modified'] >= $time_three );
+		$this->assertTrue( $tracking_data['woocommerce_allow_tracking_first_optin'] >= $time_one && $tracking_data['woocommerce_allow_tracking_first_optin'] < $time_two );
+
+		// Restore everything as it was.
```

Comment:
Just a small nitpick. We are restoring `woocommerce_allow_tracking`, but if the other two were set before this test then it's not really restoring them. Doesn't really change the tests in any way though.